### PR TITLE
[IMP] l10n_in: audit trail improvement

### PR DESCRIPTION
**Pull request by niai-odoo**
With this commit, Made the tree line clickable instead of clicking View Related Document Button. Updated the description, to be similar to chatter in audit trail tree view. Also Updated error message while deleting entries once it's posted for Indian Company.

### CODE REVIEW

**Comment by jva-odoo** [comment]
UI improvement
image (9)
image (8)

File: addons/l10n_in/models/account_invoice.py [hunk 1]
```diff
@@ -107,7 +107,7 @@ def _l10n_in_get_warehouse_address(self):
     def _unlink_l10n_in_except_once_post(self):
         # Prevent deleting entries once it's posted for Indian Company only
         if any(m.country_code == 'IN' and m.posted_before for m in self) and not self._context.get('force_delete'):
-            raise UserError(_("To keep the audit trail, you can not delete journal entries once they have been posted"))
+            raise UserError(_("To keep the audit trail, you can not delete journal entries once they have been posted.\nInstead, you can cancel the journal entry."))
```
Comment (rigr-odoo) [inline addons/l10n_in/models/account_invoice.py:110]: Don't forget to modify this in the `.pot` file :)

File: addons/l10n_in/models/mail_message.py [hunk 2]
```diff
@@ -24,8 +24,8 @@ def _compute_l10n_in_audit_log_preview(self):
                 title = message.subtype_id.display_name
             audit_log_preview = Markup("<div>%s</div>") % (title)
             for value in tracking_value_ids:
-                audit_log_preview += Markup("<li>%s <i class='o_TrackingValue_separator fa fa-long-arrow-rightmx-1 text-600' title='%s' role='img' aria-label='%s'></i>%s (%s)</li>") % (
-                    _("Changed"), _("Changed"), value._get_old_display_value()[0], value._get_new_display_value()[0], value.field.field_description)
+                audit_log_preview += Markup("<li>%s <i class='o_TrackingValue_separator fa fa-long-arrow-right mx-1 text-600' title='%s' role='img' aria-label='%s'></i>%s (%s)</li>") % (
+                    value._get_old_display_value()[0] or None, _("Changed"), value._get_old_display_value()[0], value._get_new_display_value()[0] or None, value.field.field_description)
```
Comment (rigr-odoo) [inline addons/l10n_in/models/mail_message.py:28]: `value._get_old_display_value()[0] or None` is used multiple times, perhaps a variable could be created so that the formatted would be clearer ?

File: addons/l10n_in/i18n/l10n_in.pot
Comment (rigr-odoo) [inline addons/l10n_in/i18n/l10n_in.pot:1]: Why did you delete the whole .pot file ?

Comment (niai-odoo) [reply addons/l10n_in/i18n/l10n_in.pot:1]: i haven't deleted the pot file. i just exported latest pot file.

Comment (rigr-odoo) [reply addons/l10n_in/i18n/l10n_in.pot:1]: Oh alright my bad, I saw lots of red

**Review by rigr-odoo — APPROVED** [review]

File: addons/l10n_in/models/mail_message.py [hunk 3]
```diff
@@ -24,8 +25,8 @@ def _compute_l10n_in_audit_log_preview(self):
                 title = message.subtype_id.display_name
             audit_log_preview = Markup("<div>%s</div>") % (title)
             for value in tracking_value_ids:
-                audit_log_preview += Markup("<li>%s <i class='o_TrackingValue_separator fa fa-long-arrow-rightmx-1 text-600' title='%s' role='img' aria-label='%s'></i>%s (%s)</li>") % (
-                    _("Changed"), _("Changed"), value._get_old_display_value()[0], value._get_new_display_value()[0], value.field.field_description)
+                audit_log_preview += Markup("<li>%s <i class='o_TrackingValue_separator fa fa-long-arrow-right mx-1 text-600' title='%s' role='img' aria-label='%s'></i>%s (%s)</li>") % (
+                    value._get_old_display_value()[0] or "None", arrow_title, arrow_title, value._get_new_display_value()[0] or "None", value.field.field_description)
```
Comment (william-andre) [inline addons/l10n_in/models/mail_message.py:29]: Since you use multiple times the same variables, maybe name them in order to avoid repeating them, and avoid us the trouble of having to count which argument is interpolated where?

File: addons/l10n_in/i18n/l10n_in.pot
Comment (niai-odoo) [reply addons/l10n_in/i18n/l10n_in.pot:1]: should i keep it or remove it ?

Comment (william-andre) [reply addons/l10n_in/i18n/l10n_in.pot:1]: Just remove any diff in it, lets forget about that file for now

File: addons/l10n_in/models/mail_message.py [hunk 4]
```diff
@@ -24,8 +26,8 @@ def _compute_l10n_in_audit_log_preview(self):
                 title = message.subtype_id.display_name
             audit_log_preview = Markup("<div>%s</div>") % (title)
             for value in tracking_value_ids:
-                audit_log_preview += Markup("<li>%s <i class='o_TrackingValue_separator fa fa-long-arrow-rightmx-1 text-600' title='%s' role='img' aria-label='%s'></i>%s (%s)</li>") % (
-                    _("Changed"), _("Changed"), value._get_old_display_value()[0], value._get_new_display_value()[0], value.field.field_description)
+                audit_log_preview += Markup("<li>%s <i class='o_TrackingValue_separator fa fa-long-arrow-right mx-1 text-600' title='%s' role='img' aria-label='%s'></i>%s (%s)</li>") % (
+                    value._get_old_display_value()[0] or none, arrow_title, arrow_title, value._get_new_display_value()[0] or none, value.field.field_description)
```
Comment (jco-odoo) [inline addons/l10n_in/models/mail_message.py:30]: If you are talking about using like in the chatter, then I suppose we should not put 
5.0 -> None
but 
5.0 ->

Comment (william-andre) [inline addons/l10n_in/models/mail_message.py:30]: https://github.com/odoo/odoo/pull/121242#discussion_r1201965066 hasn't been addressed?
```suggestion
                audit_log_preview += Markup(
                    "<li>%(value)s <i class='o_TrackingValue_separator fa fa-long-arrow-right mx-1 text-600' title='%(title)s' role='img' aria-label='%(title)s'></i>%(value)s (%(field)s)</li>"
                ) % {
                    'value': value._get_old_display_value()[0] or _("None"), 
                    'title': _("Changed"),
                    'field': value.field.field_description,
                }
```

There is also no need to declare the variable higher (except for minor speed improvement) because nothing is repeated.

File: addons/l10n_in/models/mail_message.py (hunk 3 above)
Comment (jco-odoo) [reply addons/l10n_in/models/mail_message.py]: With a .format then...

Comment (william-andre) [reply addons/l10n_in/models/mail_message.py]: ?
https://github.com/odoo/odoo/pull/121242#discussion_r1202413543

**Review by mart-e — APPROVED** [review]
@robodoo override=ci/security
Reason: html correctly constucted with content outside of markup call

File: addons/l10n_in/models/mail_message.py (hunk 4 above)
Comment (niai-odoo) [reply addons/l10n_in/models/mail_message.py:30]: We require it to be displayed in this manner.
5.0 -> None (Untaxed Amount)

**Review by william-andre — APPROVED** [review]
@robodoo r+
